ironfish/src/rpc/routes/chain/getTransactionStream.test.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  useAccountFixture,
+  useBurnBlockFixture,
+  useMintBlockFixture,
+} from '../../../testUtilities/fixtures'
+import { createRouteTest } from '../../../testUtilities/routeTest'
+import { GetTransactionStreamResponse } from './getTransactionStream'
+
+describe('Route chain.getTransactionStream', () => {
+  const routeTest = createRouteTest()
+
+  it('should fail if no incoming view key is specified', async () => {
+    await expect(
+      routeTest.client.request('chain/getTransactionStream', {}).waitForEnd(),
+    ).rejects.toThrow('Request failed (400) validation: incomingViewKey is a required field')
+  })
+
+  it(`should fail if block can't be found with hash`, async () => {
+    const hash = BlockHashSerdeInstance.serialize(Buffer.alloc(32, 'blockhashnotfound'))
+    const wallet = routeTest.node.wallet
+    const account = await useAccountFixture(wallet)
+
+    await expect(
+      routeTest.client
+        .request('chain/getTransactionStream', {
+          incomingViewKey: account.incomingViewKey,
+          head: hash,
+        })
+        .waitForEnd(),
+    ).rejects.toThrow(
+      `Request failed (400) validation: Block with hash ${hash} was not found in the chain`,
+    )
+  })
+
+  it('returns expected mints and burns', async () => {
+    const wallet = routeTest.node.wallet
+    const account = await useAccountFixture(wallet)
+    const asset = new Asset(account.spendingKey, 'customasset', 'metadata')
+    const stream = routeTest.client.request<GetTransactionStreamResponse>(
+      'chain/getTransactionStream',
+      {
+        incomingViewKey: account.incomingViewKey,
+      },
+    )
+    await stream.contentStream().next()
+    // Mint so we have an existing asset
+    const mintValue = BigInt(10)
+
+    const mintBlock = await useMintBlockFixture({
+      node: routeTest.node,
+      account,
+      asset,
+      value: mintValue,
+    })
+    await expect(routeTest.node.chain).toAddBlock(mintBlock)
+    // validate mint block
+    expect(await (await stream.contentStream().next()).value).toEqual(
+      expect.objectContaining({
+        transactions: expect.arrayContaining([
+          expect.objectContaining({
+            mints: expect.arrayContaining([expect.objectContaining({ value: '10' })]),
+          }),
+        ]),
+      }),
+    )
+
+    // update wallet so it sees newly added asset
+    await wallet.updateHead()
+
+    // now burn
+    const burnBlock = await useBurnBlockFixture({
+      node: routeTest.node,
+      account,
+      asset,
+      value: mintValue,
+    })
+    await expect(routeTest.node.chain).toAddBlock(burnBlock)
+    // validate burn block
+    expect(await (await stream.contentStream().next()).value).toEqual(
+      expect.objectContaining({
+        transactions: expect.arrayContaining([
+          expect.objectContaining({
+            burns: expect.arrayContaining([expect.objectContaining({ value: '10' })]),
+          }),
+        ]),
+      }),
+    )
+    stream.end()
```

Comment:
This test is hanging indefinitely, is this not the correct method to call to close the stream?